### PR TITLE
Delete API for Purchase by purchaseId

### PR DESCRIPTION
**Pull request by EXR-85**

### CODE REVIEW

File: src/main/java/com/elixr/poc/service/PurchaseService.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    public PurchaseService(PurchaseRepository purchaseRepository) {
+        this.purchaseRepository = purchaseRepository;
+    }
+
+    /*
+    Deleting the purchase from the database
+     */
+    public boolean deletePurchaseDetails(UUID purchaseId) {
+        boolean success = false;
+        try {
```
Comment (mh-sraveendran) [inline src/main/java/com/elixr/poc/service/PurchaseService.java]: Its better to check if entity exists by calling repository.findById(uuid) first.
If not found then display an error message "No matching entity found"

File: src/main/java/com/elixr/poc/service/PurchaseService.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    private DeleteResponse deleteResponse;
+
+    public PurchaseService(PurchaseRepository purchaseRepository) {
+        this.purchaseRepository = purchaseRepository;
+    }
+
+    public DeleteResponse deletePurchaseDetails(UUID purchaseId) {
+        deleteResponse = new DeleteResponse();
+        boolean purchaseRecord = purchaseRepository.existsById(purchaseId);
+        if (purchaseRecord) {
```
Comment (mh-sraveendran) [inline src/main/java/com/elixr/poc/service/PurchaseService.java]: Naming of boolean variable is not proper.
better name is "purchaseRecordExists"

File: src/main/java/com/elixr/poc/Constant/Constants.java [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package com.elixr.poc.Constant;
+
+/* this constant file holds all the error messages for all the end points */
+
+public class Constants {
```
Comment (mh-sraveendran) [inline src/main/java/com/elixr/poc/Constant/Constants.java]: Rename this class as ApplicationConstants.java

File: src/main/java/com/elixr/poc/Constant/Constants.java [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+    public static final String DELETED = "Deleted";
+    public static final String ID_MISMATCH = "The ID is not Matching";
+    public static int REQUEST_VALUE;
```
Comment (mh-sraveendran) [inline src/main/java/com/elixr/poc/Constant/Constants.java]: Dont put non final variable in Constant file

File: src/main/java/com/elixr/poc/service/PurchaseService.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+            purchaseRepository.deleteById(purchaseId);
+            deleteResponse.setSuccess(true);
+            Constants.REQUEST_VALUE = 200;
```
Comment (mh-sraveendran) [inline src/main/java/com/elixr/poc/service/PurchaseService.java]: Wrong way of sending info back to calling class.
Define a variable boolean recordAlreadyExists = false in DeleteResponse class and assign the value to that variable.

File: src/main/java/com/elixr/poc/service/PurchaseService.java (hunk 2 above)
Comment (EXR-85) [reply src/main/java/com/elixr/poc/service/PurchaseService.java]: Updated has suggested.

File: src/main/java/com/elixr/poc/Constant/Constants.java (hunk 3 above)
Comment (EXR-85) [reply src/main/java/com/elixr/poc/Constant/Constants.java]: Updated

File: src/main/java/com/elixr/poc/Constant/Constants.java (hunk 4 above)
Comment (EXR-85) [reply src/main/java/com/elixr/poc/Constant/Constants.java]: Updated.

File: src/main/java/com/elixr/poc/service/PurchaseService.java (hunk 5 above)
Comment (EXR-85) [reply src/main/java/com/elixr/poc/service/PurchaseService.java]: Updated has suggested.

**Review by mh-sraveendran — APPROVED** [review]

File: src/main/java/com/elixr/poc/rest/response/DeleteResponse.java [hunk 6]
```diff
@@ -0,0 +1,10 @@
+package com.elixr.poc.rest.response;
+
+import lombok.Data;
+
+@Data
+public class DeleteResponse {
+
+    private boolean success;
+    private String messages;
```
Comment (mh-spillai) [inline src/main/java/com/elixr/poc/rest/response/DeleteResponse.java]: It should be errorMsg. Also, could you use Jackson annotations to exclude parameters and change the parameters' label in the response?

**Review by mh-spillai — CHANGES_REQUESTED** [review]

File: build.gradle [hunk 7]
```diff
@@ -16,8 +16,14 @@ dependencies {
...
+	annotationProcessor 'org.projectlombok:lombok:1.18.24'
+
+	testCompileOnly 'org.projectlombok:lombok:1.18.24'
+	testAnnotationProcessor 'org.projectlombok:lombok:1.18.24'
 }
 
 tasks.named('test') {
 	useJUnitPlatform()
-}
+}
```
Comment (mh-spunnappala) [inline build.gradle]: Add a new line at the end of each file.

File: src/main/java/com/elixr/poc/Constant/ApplicationConstants.java [hunk 8]
```diff
@@ -0,0 +1,9 @@
+package com.elixr.poc.Constant;
+
+/* this constant file holds all the error messages for all the end points */
```
Comment (mh-spunnappala) [inline src/main/java/com/elixr/poc/Constant/ApplicationConstants.java]: Recommended comment format is /** */. Modify it in all files

File: src/main/java/com/elixr/poc/Repository/PurchaseRepository.java [hunk 9]
```diff
@@ -0,0 +1,12 @@
+package com.elixr.poc.Repository;
+
+import org.springframework.data.mongodb.repository.MongoRepository;
+import org.springframework.stereotype.Repository;
+
+import java.util.UUID;
+
+@Repository
+public interface PurchaseRepository extends MongoRepository<com.elixr.poc.data.Purchase, UUID> {
```
Comment (mh-spunnappala) [inline src/main/java/com/elixr/poc/Repository/PurchaseRepository.java]: Why the purchase class is imported with package?

**Review by mh-spunnappala — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 7 above)
Comment (EXR-85) [reply build.gradle]: Added the new line.

File: src/main/java/com/elixr/poc/Constant/ApplicationConstants.java (hunk 8 above)
Comment (EXR-85) [reply src/main/java/com/elixr/poc/Constant/ApplicationConstants.java]: Updated.

File: src/main/java/com/elixr/poc/Repository/PurchaseRepository.java (hunk 9 above)
Comment (EXR-85) [reply src/main/java/com/elixr/poc/Repository/PurchaseRepository.java]: I removed now.

File: src/main/java/com/elixr/poc/rest/response/DeleteResponse.java (hunk 6 above)
Comment (EXR-85) [reply src/main/java/com/elixr/poc/rest/response/DeleteResponse.java]: Updated.

**Review by mh-spunnappala — APPROVED** [review]

**Pull request merged**